packages/node/test/machine/integration/protocols/protocols.spec.ts

```diff
@@ -0,0 +1,112 @@
+import AppWithAction from "@counterfactual/cf-adjudicator-contracts/build/AppWithAction.json";
+import { OutcomeType } from "@counterfactual/types";
+import { Contract, ContractFactory, Wallet } from "ethers";
+import { BaseProvider } from "ethers/providers";
+import { bigNumberify } from "ethers/utils";
+
+import { CONVENTION_FOR_ETH_TOKEN_ADDRESS } from "../../../../src/constants";
+import { Protocol } from "../../../../src/machine";
+import { toBeEq } from "../bignumber-jest-matcher";
+import { connectToGanache } from "../connect-ganache";
+
+import { TestRunner } from "./test-runner";
+
+let wallet: Wallet;
+let appWithAction: Contract;
+
+expect.extend({ toBeEq });
+
+enum ActionType {
+  SUBMIT_COUNTER_INCREMENT,
+  ACCEPT_INCREMENT
+}
+
+beforeAll(async () => {
+  [{}, wallet, {}] = await connectToGanache();
+
+  appWithAction = await new ContractFactory(
+    AppWithAction.abi,
+    AppWithAction.bytecode,
+    wallet
+  ).deploy();
+});
+
+describe("Three mininodes", () => {
+  it("Can run all the protocols", async () => {
```

Comment:
This  test can be broken up by protocol and edit `describe` level variables along the way; then if any particular part fails we will have a more clear idea as to what the failure was